src/js/component.js

```diff
@@ -201,12 +201,25 @@ vjs.Component.prototype.createEl = function(tagName, attributes){
   return vjs.createEl(tagName, attributes);
 };
 
+/**
+ * Return a localised string if available
+ *
+ * @param  {String} string  String to be localised
+ * @return {String}
+ */
 vjs.Component.prototype.localize = function(string){
-  var lang = this.player_.language(),
-      languages = this.player_.languages();
-  if (languages && languages[lang] && languages[lang][string]) {
+  var lang = ('' + this.player_.language()).toLowerCase();
+  var languages = this.player_.languages();
+  if (!languages) {
+    return string;
+  }
+  if (languages[lang] && languages[lang][string]) {
     return languages[lang][string];
   }
+  var primaryCode = lang.split('-')[0];
```

Comment:
Though I think no one likes it at this point, most of the rest of the code declares variables at the top of the scope.